### ISSUE TITLE
Improve logging on null exception when creating an instance

### ISSUE THREAD

**Issue by BenoitDuffez** [InsertKoinIO/koin#283]
Context: Koin 1.0.0 on Android. Injecting a ViewModel created with one parameter as constructor argument. Said parameter is also injected and contains fields that start off nulls.

Pseudo code was:

```
class MyViewModel(val whatever: Whatever): ViewModel {
    val bogusItem = whatever.foo
    ...
}
```

Module was:
```
val whatever = module {
    single<Whatever>()
    viewModel<MyViewModel>()
}
```

When run, the exception is:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{my.app/my.calling.Activity}: org.koin.error.BeanInstanceCreationException: Can't create definition for 'Factory [name='MyViewModel',class='com.whatever.MyViewModel', binds~(androidx.lifecycle.ViewModel)]' due to error :
        null
        java.lang.reflect.Constructor.newInstance0(Native Method)
        java.lang.reflect.Constructor.newInstance(Constructor.java:334)
```

The goal of this issue is to improve the "null" here which is a bit confusing.

I ran the code with a breakpoint at `InstanceHolder.create` in the catch block. There's a Kotlin null pointer exception.

I re-read my code and in fact the `whatever field` is using the dependency too soon. I fixed it by changing my view model as:

```
...
val foo
   get() = whatever.foo
...
```

Then it works.

Now I know that this is absolutely not a bug in Koin, but I figured that maybe the error message could be better. Like if we detect a NPE, maybe add a message like "It seems that the injected dependency ran against a NPE, please check your properties for possible early accesses to uninstantiated dependencies".

**Comment by IndexOverflow** [comment]
We're seeing this quite a lot too. Would be really helpful if the produced error could be more informative.

**Comment by arnaudgiuliani** [comment]
Still there in `1.0.2`?

**Comment by IndexOverflow** [comment]
> Still there in `1.0.2`?

Just tried to upgrade. Seeing it there as well. We primarily see the null error if a dependency to be injected cannot be created (i.e a config loader which throws an Exception).

**Comment by arnaudgiuliani** [comment]
I think I've fixed it in `2.0.0` but I have to check it :/ 
stay tunned

**Comment by arnaudgiuliani** [comment]
please check next `2.0.0` version
